### ISSUE TITLE
otb format

### ISSUE THREAD

**Issue by irenicus30** [otland/forgottenserver#2769]
Why use otb format? I see that otb format is used in two situations: 
1. tibia map description
2. tibia items description

Ok I can understand that real tibia map requires some binaryformat like otb becosue real tibia map is very big and binary format gives performance boost comparing to xml, But items in otb format - is it only becouse like 10 years ago decided to specify items both in xml and otb? (first commit is 2013 and this is fork of another opentibia server)? Maybe good idea would be to refactor items so that it does not require otb binary format.

PS note that this is only a question not actual proposal to refactor anything becouse it might be too early

**Comment by Sickness-chan** [comment]
OTB is absolutely useless, it even complicates a lot of things because client & server ID's are different, for which there's absolutely no reason. Everything could be done in the items.xml file.

**Comment by irenicus30** [comment]
Hi thanks for reply. Maybe it could be possible to allow alternative: (only xml format with some extra fields) or (xml and otb format to allow compatibility with existing datapacks)?

**Comment by EPuncker** [comment]
I remember someone saying that the id mismatch is to keep "consistency" or something like this, maybe @marksamman can tell more about (or maybe I dreamed about it and I'm completely wrong lol), and about the format itself: 
read #2458 #2206 #2272 we maybe will see a new format in a few years tho 😋

**Comment by dudantas** [comment]
> OTB is absolutely useless, it even complicates a lot of things because client & server ID's are different, for which there's absolutely no reason. Everything could be done in the items.xml file.

Server id was a real failure in open tibia.

**Comment by gesior** [comment]
10 years ago it wasn't failure. It was thing required to make OTS easily change from version to version.

If you decide 'i want to make 8.6 ots', it would be easier to use client IDs on server.

If you want to upgrade your server from version to version, you get risk that tibia owners will mix IDs to make your life hell. For that case there was a tool that traced items by their sprite images and generate consistent client id -> server id map.

**Comment by giuinktse7** [comment]
> 10 years ago it wasn't failure. It was thing required to make OTS easily change from version to version.
> 
> If you decide 'i want to make 8.6 ots', it would be easier to use client IDs on server.
> 
> If you want to upgrade your server from version to version, you get risk that tibia owners will mix IDs to make your life hell. For that case there was a tool that traced items by their sprite images and generate consistent client id -> server id map which was saved as items.otb.

For reference, a version of the item tracing is in [ottools/ItemEditor](https://github.com/ottools/ItemEditor) here: https://github.com/ottools/ItemEditor/blob/2bccd9e5396e742c2403658281b85b4c577570db/Source/PluginInterface/Item.cs#L291

I think the sprite hashes present in the current items.otb in this repository (forgottenserver) are generated using this function.

**Comment by Zbizu** [comment]
With the introduction of assets, the tracking is even more necessary. We don't know when or how will the structure move in the future. Client ids have long history of moving. We used to have old borders in low client ids, now we have jester outfit quest assets in that place.

**Comment by giuinktse7** [comment]
True. Also noteworthy is that some items have received sprite touchups in recent years (no clue how many). For these items, the sprite hashes will change despite the "underlying item type" not having changed. So the sprite hashes are not bulletproof, but they do help keep some parity between versions.
